Calculator.js

```diff
@@ -1,6 +1,184 @@
 var Calculator = function () {
-  this.operation = []
-  this.value = []
+  this.operations = []
+  this.operands = []
+  this.display = '0'
+  this.displayFlag = true
 }
 
-module.exports = Calculator
+Calculator.prototype = {
+
+  // Functions which interact with external interface
+
+  acceptInput: function ( value ) {
+    if ( value === null ) {
+      return
+    }
+    this.value = value
+    this.handleInput()
+  },
+
+  displayScreen: function () {
+    this.displayFlag = true
+    if ( this.display === '' ) {
+      this.display = '0'
+    } else if ( this.display === '0' ) {
+      this.display = this.value
+    } else {
+      this.display += this.value
+    }
+  },
+
+  getDisplay: function () {
+    return this.display
+  },
+
+  getDisplayLength: function () {
+    return this.display.length
+  },
+
+  shouldDisplay: function () {
+    return this.displayFlag
+  },
+
+  // Special operation functions
+
+  allClear: function () {
+    this.operands = []
+    this.operations = []
+    this.display = '0'
+    this.displayFlag = true
+    this.value = '0'
+    return this.displayScreen()
+  },
+
+  changeSign: function () {
+    if ( !this.checkForSignInDisplay() ) {
+      this.display = '-' + this.display
+    } else {
+      this.display = this.display.slice( -( this.getDisplayLength() - 1 ) )
+    }
+  },
+
+  computePercentage: function () {
+    this.display = ( this.display / 100 )
```

Comment:
you're relying on type casting here which can do the wrong thing sometimes. Consider always storing your display as a number instead of a string.